src/sentry/api/endpoints/organization_member_team_details.py

```diff
@@ -103,6 +123,23 @@ def _create_access_request(
 
         omt.send_request_email()
 
+    def get(
+        self,
+        request: Request,
+        organization: Organization,
+        member: OrganizationMember,
+        team_slug: str,
+    ) -> Response:
+        try:
+            team = Team.objects.get(organization=organization, slug=team_slug)
+        except Team.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if not OrganizationMemberTeam.objects.filter(team=team, organizationmember=member).exists():
+            raise ResourceDoesNotExist
+
+        return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=200)
```

Comment:
Following up on https://github.com/getsentry/sentry/pull/32498#discussion_r833342190 from @markstory:

> Do we need the `projects` in this response? I thought we were trying to move away from automatically including 1:N relations.

In practical terms, no. I believe only the Team Members view uses this endpoint, and when I replace all instances of `TeamWithProjectsSerializer` with `TeamSerializer`, the code that calls the existing POST and DELETE methods continues to work fine. The endpoint is undocumented, so I think there's no abstract harm in changing it.

However, I could use some advice on front-end best practices before I make the change. The [code in `teamMembers.tsx`](https://github.com/getsentry/sentry/blob/bb42837000875ecebc4276a0d6b4a68b30375eb4/static/app/views/settings/organizationTeams/teamMembers.tsx#L49) that unpacks the response does so with the `Member` type, which [has `projects` as an attribute](https://github.com/getsentry/sentry/blob/496e21347d491f2d06bc0b6eac5a08c9b2ec00a5/static/app/types/organization.tsx#L102).

If we remove projects from the response, should we also change the front-end type? Is it worth it?

Reply:
without understanding fully what this serializer does, 

I'd assume yes - you would probably want to remove projects from the types file.
TS would shout at you and help point out if anythings depending on that type being available (and likely suggest you to clean up some FE flows if there are)

If there are dependencies, probably good to remove the FE deps first and ensure nothings expecting `projects`, then remove it from the api response 🤷

Reply:
Agree with Nathan that removing the front-end attribute from the types first will help identify any UI flows that need to change, ship those changes and then deploy the endpoint changes.

Reply:
👍 Will try removing projects from the type.

I also want to see if we can simplify the whole thing by serializing the `OrganizationMemberTeam` rather than the entire `Team`. I believe `teamMembers.tsx` is the only thing consuming the POST response and I don't think it needs all of it.

Reply:
It dawned on me that this endpoint used to be only a POST method and was reflecting the state of the parent object (the team). Since GET and PUT are brand new, I was making a mistake by trying to have their response contract match POST. (Further complicating things was the DELETE body, which is just unnecessary.)

As of 11bc4a1fd74c0d7b5fd8b93b2e436ce645035bdb, GET and PUT use the newly introduced `OrganizationMemberTeamDetailsSerializer`. It's very minimal and we can always add more.

It might still be worth taking the projects out of the POST response, but that's a preexisting issue that we can handle separately.